middleware/logging.go

```diff
@@ -32,7 +32,7 @@ func (l Log) Wrap(next http.Handler) http.Handler {
 		}
 		i := &interceptor{ResponseWriter: w, statusCode: http.StatusOK}
 		next.ServeHTTP(i, r)
-		if 100 <= i.statusCode && i.statusCode < 400 {
+		if 100 <= i.statusCode && (i.statusCode < 400 || i.statusCode < 429) {
```

Comment:
If we're going to go with this approach, I'd rather we use Debug level for _all_ 400s. It's unlikely that an operator would guess that we log all 400s other than 429s.